torch/_inductor/decomposition.py

```diff
@@ -229,11 +229,22 @@ def mm(self, input2):
         guard_size_oblivious,
     )
 
-    # Our matrix vector multiplies only achieve peak bandwidth with coordinate descent tuning.
-    # todo: Look into why and fix it (hopefully)
-    if config.coordinate_descent_tuning:
-        if self.shape[0] == 1 or input2.shape[1] == 1:
-            return (self.unsqueeze(2) * input2.unsqueeze(0)).sum(dim=1)
+    def mul_sum_decomp(self, input2):
+        return (self.unsqueeze(2) * input2.unsqueeze(0)).sum(dim=1)
+
+    def coordesc_or(cond):
+        return config.coordinate_descent_tuning or cond
+
+    if config.decompose_mm_to_mv:
+        gso = guard_size_oblivious
+        # Corresponds to BS=1 cases morally. If it's contiguous we can decompose it
+        # and generate efficiently kernels without autotuning. Otherwise we need coordinate descent tuning
+        if gso(self.shape[0] == 1) and coordesc_or(gso(input2.stride(0) == 1)):
```

Comment:
Feel like if it's an unbacked symint here we probably dont want to decompose to mv, since that could be any shape and potentially large regression.  I think we'd want definitely_true